### PR TITLE
Centralizes definitions of accumulo tables

### PR DESCRIPTION
**Pull request by keith-turner**
Three tables in the "accumulo." namespace were defined as constants in the three classes in the code. The change gathers the definitions of these three tables into a single enum in the code. The motivation for this change was that the new fate code introduced in #4133 needed to know of the all the accumulo tables ids.  The way the code was structured if a new accumulo.X table is added in the future someone may not realize the fate code needs to know about it.  This change makes the dependency more apparent in the code.

Existing constants for the metadata and root table name and id were not inlined in the commit for two reasons.  First, it would make this commit hard to review.  Second, it would make merging main into the elasticity branch more difficult (its already tricky).  Inlining these constants can easily be done as a stand alone commit later when the elasticity branch is ready to merge into main.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/fate/FateInstanceType.java [hunk 1]
```diff
@@ -18,20 +18,20 @@
...
-import org.apache.accumulo.core.metadata.MetadataTable;
-import org.apache.accumulo.core.metadata.RootTable;
+import org.apache.accumulo.core.metadata.AccumuloTables;
 
 public enum FateInstanceType {
   META, USER;
 
-  private static final Set<TableId> META_TABLES =
-      Set.of(RootTable.ID, MetadataTable.ID, FateTable.ID);
+  private static final Set<TableId> META_TABLES = Arrays.stream(AccumuloTables.values())
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/fate/FateInstanceType.java:33]: This bit of code was the motivation for this change.

**Comment by EdColeman** [comment]
Would it help merging forward if the AccumuloTables enum and just those related changes were targeted for 3.1?  The changes that are not directly retreaded to elasticity seem to be general enough and an improvement that maybe they could be in 3.1?

**Comment by keith-turner** [comment]
@EdColeman and @ddanielr for making the change in main, thinking of the following strategy.  

 * Create just the new AccumuloTables enum in main and merge it forward
 * Do the refactor in main to inline and merge that commit forward ignoring the changes in it
 * Do the refactor in elasticity as its own commit

If that strategy sounds good I can do that w/o any further PRs.

**Comment by ddanielr** [comment]
> @EdColeman and @ddanielr for making the change in main, thinking of the following strategy.
> 
> * Create just the new AccumuloTables enum in main and merge it forward
> * Do the refactor in main to inline and merge that commit forward ignoring the changes in it
> * Do the refactor in elasticity as its own commit
> 
> If that strategy sounds good I can do that w/o any further PRs.

I think that's an easier strategy and keeps the elasticity complexity fixes from being hidden in a merge commit.

**Comment by keith-turner** [comment]
Looks like elasticity is not currently up to do date with main.  Working on that merge first.

**Comment by keith-turner** [comment]
Made multiple commits to add the new AccumuloTable enum to main and elasticity.

 * Added the enum in main in 0d5a87cedddc83a17851c99efa79c642b9bbc4e3
 * Merged the new enum into elasticity in 468e7fe3f83f76db901d94ab8c0a8d2a73dc0590
 * Did the inline refactoring in main in da7866ee3606fd43202f536fd50a6e2d890a1dee
 * Did the inline refactoring in elasticity in 90fcc0be17fe142ccb0de08363820d73da14abbb
 * Merged -sours main into elasticity in ff98ee4498b39838d781065b4202508ae50d36cc

**Comment by cshannon** [comment]
Just catching up on this now, this LGTM and makes a lot of sense, thanks for refactoring this.

**Pull request merged**